internal/cache/cache.go

```diff
@@ -0,0 +1,69 @@
+package cache
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+)
+
+type Cache struct {
+	LastTimestamp string `json:"lastTimestamp"`
```

Comment:
what do you think to add the `directoryPath ` and `resourceName ` as part of the struct and create a New() func to initialize the cache? , i think this also could help on reducing some amount of parameteres in other functions